scipy/sparse/construct.py

```diff
@@ -23,6 +24,9 @@
 from .base import issparse
 
 
+np.random = random_patched
```

Comment:
I'm not sure this approach will work. If a user passes `np.random.RandomState` in explicitly with older numpy versions, they won't get the patched `randint` and will encounter an error.

I'd recommend having `_numpy_compat` define a shim that accepts the random state as an argument, and avoid patching modules which can be easy to get wrong.